dataflux_pytorch/dataflux_iterable_dataset.py

```diff
@@ -111,53 +114,66 @@ def __init__(
                 during initialization.
         """
         super().__init__()
-        self.storage_client = storage_client
-        if not storage_client:
-            self.storage_client = storage.Client(project=project_name, )
-        dataflux_core.user_agent.add_dataflux_user_agent(self.storage_client)
+        self.storage_client = None
+        if OS == LINUX:
+            self.storage_client = (
+                storage_client
+                if storage_client is not None
+                else storage.Client(project=project_name)
+            )
+            dataflux_core.user_agent.add_dataflux_user_agent(self.storage_client)
         self.project_name = project_name
         self.bucket_name = bucket_name
-        self.data_format_fn = data_format_fn
+        self.data_format_fn = (
+            data_format_fn if data_format_fn is not None else self.data_format_default
```

Comment:
If the user passes a custom data_format_fn which is a lambda, will it fail to pickle it? If so, should we add a doc comment or something that on non-Linux platforms they need to pass a function definition rather than a lambda?

Reply:
Lambda is definitely preferable for readability unless you ran into specific errors with passing a lambda.